tests/integration/Segments/DynamicSegments/Filters/WooCommerceNumberOfOrdersTest.php

```diff
@@ -0,0 +1,115 @@
+<?php
+
+namespace MailPoet\Segments\DynamicSegments\Filters;
+
+use Carbon\Carbon;
+use MailPoet\Entities\DynamicSegmentFilterData;
+use MailPoet\Entities\DynamicSegmentFilterEntity;
+use MailPoet\Entities\SegmentEntity;
+use MailPoet\Entities\SubscriberEntity;
+
+class WooCommerceNumberOfOrdersTest extends \MailPoetTest {
+
+  private $numberOfOrders;
+
+  private $orders;
+
+  public function _before() {
+    $this->numberOfOrders = $this->diContainer->get(WooCommerceNumberOfOrders::class);
+    $this->cleanUp();
+
+    $userId1 = $this->tester->createWordPressUser('customer1@example.com', 'customer');
+    $userId2 = $this->tester->createWordPressUser('customer2@example.com', 'customer');
+    $userId3 = $this->tester->createWordPressUser('customer3@example.com', 'customer');
+
+    $this->orders[] = $this->createOrder(['user_id' => $userId1, 'post_date' => Carbon::now()->subDays(3)->toDateTimeString()]);
+    $this->orders[] = $this->createOrder(['user_id' => $userId2]);
+    $this->orders[] = $this->createOrder(['user_id' => $userId2]);
+    $this->orders[] = $this->createOrder(['user_id' => $userId3]);
+  }
+
+  public function testItGetsCustomersThatPlacedTwoOrdersInTheLastDay() {
+    $segmentFilter = $this->getSegmentFilter('=', 2, 1);
+    $queryBuilder = $this->numberOfOrders->apply($this->getQueryBuilder(), $segmentFilter);
+    $result = $queryBuilder->execute()->fetchAll();
+    expect(count($result))->equals(1);
+    $subscriber1 = $this->entityManager->find(SubscriberEntity::class, $result[0]['inner_subscriber_id']);
+    assert($subscriber1 instanceof SubscriberEntity);
```

Comment:
Rosta told me that you told him that it is better to use `expect($subscriber1)->isInstanceOf(SubscriberEntity::class);` rather than `assert`. So why are you using `assert` in this test, is there a benefit here?

Reply:
I think `expect($subscriber1)->isInstanceOf(SubscriberEntity::class);` still doesn't work with PHPStan. `$this->asertInstanceOf` was the one with nicer error mesages.

Reply:
Yeah, the error message of `$this->assertInstanceOf()` is more detailed and precise than the error message of `assert()`. I used `assert()` here because I based those tests on the UserRoleTest tests, and it didn't occur to me to replace `assert()` with `$this->assertInstanceOf()`. I pushed a commit that replaces all the generic assertions with more specific ones, not just for the case that you mentioned here.